src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/BulkCopyTimeoutTest.java

```diff
@@ -58,7 +58,9 @@ public void execute() throws SQLException {
 
     private void testBulkCopyWithTimeout(int timeout) throws SQLException {
         BulkCopyTestWrapper bulkWrapper = new BulkCopyTestWrapper(connectionString);
-        bulkWrapper.setUsingConnection((0 == ThreadLocalRandom.current().nextInt(2)) ? true : false);
+        bulkWrapper.setUsingConnection((0 == ThreadLocalRandom.current().nextInt(2)) ? true : false, ds);
+        bulkWrapper.setUsingXAConnection((0 == ThreadLocalRandom.current().nextInt(2)) ? true : false, dsXA);
+        bulkWrapper.setUsingPooledConnection((0 == ThreadLocalRandom.current().nextInt(2)) ? true : false, dsPool);
```

Comment:
not a big deal since this is only a test, but ThreadLocalRandom.current() can be factored out so it's only called once